nixos/modules/config/system-libs.nix

```diff
@@ -0,0 +1,114 @@
+# This module defines the packages that appear in
+# /run/current-system/sw.
+
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.libraries;
+
+  pathsToLink = [ "/lib" "/share" ];
+  mkLibPath = pkgs: pkgs.buildEnv {
+    name = "libraries";
+    paths = cfg.unsafePackages pkgs;
+    inherit pathsToLink;
+    ignoreCollisions = true;
+  };
+  mkSanitized = pkgs: pkgs.buildEnv {
+    name = "sanitized-libraries";
+    paths = cfg.packages pkgs;
+    inherit pathsToLink;
+    ignoreCollisions = true;
+    postBuild = ''
+      for i in $out/lib/*; do
+        if [ -f "$(readlink -f "$i")" ]; then
+          rm "$i"
+        fi
+      done
+    '';
+  };
+
+  libPath = mkLibPath pkgs;
+  libPath32 = mkLibPath pkgs.pkgsi686Linux;
```

Comment:
Having both implies that installing lib32 support would cause the system to compile an x86 toolchain in addition to the existing x86_64 toolchain.  This sounds like way too much overhead to me. Couldn't such package already provide a lib32 directory as part of the outputs of the package?